### ISSUE TITLE
useStrapiApp hook with ImageDialog throw error in Strapi 5 plugin

### ISSUE THREAD

**Issue by JakubVavra** [strapi/strapi#21957]
Hi, in my Strapi 5 plugin a would like to access media library. I looked into your wysiwyg field (Markdown rich text field) and you use useStrapiApp like this.

```
const components = useStrapiApp('ImageDialog', (state) => state.components);
const MediaLibraryDialog = components['media-library'];
```

But when I try this in Strapi 5 plugin in typescript, Strapi throw this error.

![image](https://github.com/user-attachments/assets/e91665aa-63f6-4e4f-b396-66bfa16a78c4)

Can someone help me?

Thanks

**Comment by mattmilburn** [comment]
This is the same issue reported in https://github.com/strapi/strapi/issues/22162

**Comment by JakubVavra** [comment]
No, but you can try this plugin. It add custom field for rich text.

https://www.npmjs.com/package/@sklinet/strapi-plugin-tinymce

**Comment by maruthasalamr** [comment]
No,tinymce is paid so i'm trying to use ckeditor plugin. 

```
 app.customFields.register({
        name: "Wysiwyg",
        pluginId: pluginId,
        type: "string",
        intlLabel: {
          id: `${pluginId}.plugin.name`,
          defaultMessage: name,
        },
        intlDescription: {
          id: `${pluginId}.plugin.description`,
          defaultMessage: 'Edit the text input',
        },
        components:{
          Input: async () => import("./components/Wysiwyg/index"),
        }
      });
      app.registerPlugin({
        id: pluginId,
        initializer: Initializer,
        isReady: false,
        name,
      });
```
if i use customFields it's adding as a new custom field in content type builder but when i use like this 

```
app.addFields({ type: 'wysiwyg', Component: Wysiwyg });
```

default richtext is not able to replace with the ckeditor
so can you help to fix ? @JakubVavra

**Comment by duydb** [comment]
> Hi, in my Strapi 5 plugin a would like to access media library. I looked into your wysiwyg field (Markdown rich text field) and you use useStrapiApp like this.
> 
> ```
> const components = useStrapiApp('ImageDialog', (state) => state.components);
> const MediaLibraryDialog = components['media-library'];
> ```
> 
> But when I try this in Strapi 5 plugin in typescript, Strapi throw this error.
> 
> ![image](https://github.com/user-attachments/assets/e91665aa-63f6-4e4f-b396-66bfa16a78c4)
> 
> Can someone help me?
> 
> Thanks

https://docs.strapi.io/cms/plugins-development/create-a-plugin#configuration-with-a-local-plugin

To resolve the issue, remove @strapi/strapi as a dev dependency from your plugin. This ensures that your plugin uses the same instance of Strapi’s core modules as the main application, preventing conflicts and the associated errors.
